cli/util/options.js

```diff
@@ -138,3 +140,65 @@ function help(config, options) {
 }
 
 exports.help = help;
+
+/** Merges two sets of options into one, preferring the current over the parent set. */
+function merge(config, currentOptions, parentOptions) {
+  const mergedOptions = {};
+  for (const [key, { mutuallyExclusive }] of Object.entries(config)) {
+    if (currentOptions[key] == null) {
+      if (parentOptions[key] != null) {
+        // only parent value present
+        if (Array.isArray(parentOptions[key])) {
+          let exclude;
+          if (mutuallyExclusive != null && (exclude = currentOptions[mutuallyExclusive])) {
+            mergedOptions[key] = parentOptions[key].filter(value => !exclude.includes(value));
+          } else {
+            mergedOptions[key] = parentOptions[key].slice();
+          }
+        } else {
+          mergedOptions[key] = parentOptions[key];
+        }
+      }
+    } else if (parentOptions[key] == null) {
+      // only current value present
+      if (Array.isArray(currentOptions[key])) {
+        mergedOptions[key] = currentOptions[key].slice();
+      } else {
+        mergedOptions[key] = currentOptions[key];
+      }
+    } else {
+      // both current and parent values present
+      if (Array.isArray(currentOptions[key])) {
+        let exclude;
+        if (mutuallyExclusive != null && (exclude = currentOptions[mutuallyExclusive])) {
+          mergedOptions[key] = [
+            ...currentOptions[key],
+            ...parentOptions[key].filter(value => !currentOptions[key].includes(value) && !exclude.includes(value))
+          ];
+        } else {
+          mergedOptions[key] = [
+            ...currentOptions[key],
+            ...parentOptions[key].filter(value => !currentOptions[key].includes(value)) // dedup
+          ];
+        }
+      } else {
+        mergedOptions[key] = currentOptions[key];
```

Comment:
Another improvement is fallback to `mergedOptions[key] = currentOptions[key]` by default due to it common for all cases like:
```js
for (const [key, { mutuallyExclusive }] of Object.entries(config)) {
  if (currentOptions[key] == null) {
    if (parentOptions[key] != null) {
      if (Array.isArray(parentOptions[key])) {
        ...
        constinue;
      }
    }
  } else if (...) {
     if (Array.isArray(currentOptions[key])) {
       ...
       constinue;
     }
  } else {
    if (Array.isArray(currentOptions[key])) {
      ...
      constinue;
    }
  }
  mergedOptions[key] = currentOptions[key];  // use as fallback
}
```

Reply:
Hmm, but wouldn't continue skip the fallback here?

Reply:
Checked this locally, but when looking at the code it seems significantly harder to understand, also because there's a new `} else { continue; }` branch then in the first case. Would prefer to keep it the way it is for clarity.

Reply:
ok